### PR TITLE
FEDX-651 : Migrate w_dio repo to GHA

### PR DESCRIPTION
**Pull request by dustinlessard-wf**
### New Pull Request Checklist

- [ ] I have read the [Documentation](https://pub.dev/documentation/dio/latest/)
- [ ] I have searched for a similar pull request in the [project](https://github.com/cfug/dio/pulls) and found none
- [ ] I have updated this branch with the latest `main` branch to avoid conflicts (via merge from master or rebase)
- [ ] I have added the required tests to prove the fix/feature I'm adding
- [ ] I have updated the documentation (if necessary)
- [ ] I have run the tests without failures
- [ ] I have updated the `CHANGELOG.md` in the corresponding package

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      - uses: dart-lang/setup-dart@v1.6.2
+        with:
+          sdk: ${{ matrix.sdk }}
+      - name: Install tools
+        run: |
+          sudo apt-get update -qq && \
+          sudo apt-get dist-upgrade -y && \
+          sudo apt-get install -y jq && \
+          sudo apt-get autoremove -y && \
+          sudo apt-get clean all
```
Comment (robbecker-wf) [inline .github/workflows/ci.yaml]:
```suggestion
```
We don't need to do any of this in GHA anymore. We can remove the whole build-image job

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,138 @@
+name: CI
```
Comment (robbecker-wf) [inline .github/workflows/ci.yaml:1]: Actually now that it exists, let's just use the workflows from https://github.com/Workiva/gha-dart-oss

**Review by robbecker-wf — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (robbecker-wf) [reply .github/workflows/ci.yaml:1]: Well, actually, those workflows assume dart_dev is installed, so maybe we can't use those here

**Comment by robbecker-wf** [comment]
I'm not seeing the pub artifacts attached anywhere.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+          cp -r json_serializable-3.5.2/build.yaml json_serializable/
+          cp -r json_serializable-3.5.2/pubspec.yaml json_serializable/
+          echo "tar czf pub_package.pub.tgz"
+          tar czf pub_package.pub.tgz -C json_serializable .
+
+      - name: Upload artifacts for json_annotation-3.1.1
+        uses: actions/upload-artifact@v4
+        with:
+          name: json_annotation-3.1.1-artifact
+          path: pub_temp/*
```
Comment (dustinlessard-wf) [inline .github/workflows/ci.yaml]: This doesn't seem right.

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+      - uses: actions/checkout@v4
+      - uses: dart-lang/setup-dart@v1.6.2
+        with:
+          sdk: ${{ matrix.sdk }}
+
+      - name: Install dependencies
+        run: |
+          cd json_annotation-3.1.1
+          dart pub get
+          cd ..
```
Comment (robbecker-wf) [inline .github/workflows/ci.yaml:35]: After we install dependencies, can we do the SBOM generation here?

**Review by robbecker-wf — APPROVED** [review]
+10 CI passing, SBOM attached, tests run and attached.

**Comment by robbecker-wf** [comment]
@Workiva/release-management-p will you merge with WKB still on here? Let's see.

**Pull request closed without merging**